CRM/Core/BAO/CustomQuery.php

```diff
@@ -387,20 +387,27 @@ public function where() {
 
               // CRM-14563,CRM-16575 : Special handling of multi-select custom fields
               if ($isSerialized && !empty($value) && !strstr($op, 'NULL') && !strstr($op, 'LIKE')) {
+                $sp = CRM_Core_DAO::VALUE_SEPARATOR;
                 if (strstr($op, 'IN')) {
-                  $value = str_replace(",", "[[:cntrl:]]*|[[:cntrl:]]*", $value);
+                  $value = str_replace(",", "$sp|$sp", $value);
                   $value = str_replace('(', '[[.left-parenthesis.]]', $value);
                   $value = str_replace(')', '[[.right-parenthesis.]]', $value);
                 }
                 $op = (strstr($op, '!') || strstr($op, 'NOT')) ? 'NOT RLIKE' : 'RLIKE';
-                $value = "[[:cntrl:]]*" . $value . "[[:cntrl:]]*";
+                $value = $sp . $value . $sp;
                 if (!$wildcard) {
```

Comment:
@monishdeb I was just hoping you would remember how those regex chars work since despite us having spent what seems like hours making sense of it the mere sight of that regex sent my brain to mush when I saw it :-).

Needless to say I highly recommend unit tests - there is an api that can be used for tests on reports..... used in api_v3_ReportTemplateTest